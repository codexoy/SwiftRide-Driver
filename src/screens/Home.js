import React from "react"
import { View, Text, Button, StyleSheet } from "react-native"
import { logout } from "../actions/auth"
import { connect } from "react-redux"

class Home extends React.Component {
	constructor(props) {
		super(props)
		this.logout = this.logout.bind(this)
	}

	logout() {
		this.props.dispatch(
			logout(() => {
				this.props.navigation.navigate("Auth")
			})
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<Text>Hi!</Text>
				<Button onPress={this.logout} title="התנתק" />
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})

export default connect()(Home)
