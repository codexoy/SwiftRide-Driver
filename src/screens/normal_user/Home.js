import React from "react"
import { View, StyleSheet } from "react-native"
import { logout } from "../../actions/auth"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import SuccessModal from "../../components/SuccessModal"
import { signUpRoles } from "../../consts"
import { deleteDeviceToken } from "../../actions/utils"

export class Home extends React.Component {
	constructor(props) {
		super(props)
		this.logout = this.logout.bind(this)
		this.role = "normal"
		if (this.props.user.hasOwnProperty("teacher_id")) {
			// it's a teacher
			this.role = signUpRoles.teacher
		} else if (this.props.user.hasOwnProperty("my_teacher")) {
			// it's a student
			this.role = signUpRoles.student
		}
	}

	async logout() {
		await this.props.dispatch(deleteDeviceToken())
		this.props.dispatch(
			logout(async () => {
				this.props.navigation.navigate("Auth")
			})
		)
	}
	render() {
		return (
			<View style={styles.container}>
				<SuccessModal
					animationType="none"
					visible={true}
					image="pending"
					title={strings("normal_user.home.pending")}
					desc={strings(
						"normal_user.home." + this.role + "_pending_desc"
					)}
					buttonPress={() => {
						this.logout()
					}}
					button={strings("settings.logout")}
				/>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})
const mapStateToProps = state => {
	return {
		user: state.user
	}
}

export default connect(mapStateToProps)(Home)
