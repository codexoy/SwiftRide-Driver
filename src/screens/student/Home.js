import React, { Fragment } from "react"
import {
	ScrollView,
	View,
	Text,
	Image,
	StyleSheet,
	TouchableHighlight,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import ShadowRect from "../../components/ShadowRect"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import LessonPopup from "../../components/LessonPopup"
import { getPayments } from "../../actions/lessons"
import { MAIN_PADDING } from "../../consts"
import StudentPayments from "../../components/StudentPayments"
import StudentNextLessonView from "../../components/StudentNextLessonView"

export class Home extends React.Component {
	static navigationOptions = () => {
		return {
			title: "home",
			tabBarLabel: strings("tabs.home"),
			tabBarAccessibilityLabel: strings("tabs.home"),
			tabBarTestID: "HomeTab"
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			lesson: "",
			payments: [],
			lessonPopupVisible: false,
			loading: true
		}

		this._handleRequests()
	}

	_handleRequests = async () => {
		await this._getLesson()
		await this._getPayments()
		this.setState({ loading: false })
	}
	_navigateToSettings = () => {
		this.props.navigation.navigate("Settings")
	}

	_getLesson = async () => {
		const now = new Date().toISOString()
		const resp = await this.props.fetchService.fetch(
			"/lessons/?limit=1&is_approved=true&date=ge:" + now,
			{ method: "GET" }
		)
		if (resp.json["data"].length == 0) return
		this.setState({
			lesson: resp.json["data"][0]
		})
	}

	_getPayments = async () => {
		const payments = await getPayments(this.props.fetchService)
		this.setState({
			payments: payments.payments
		})
	}

	lessonPress = () => {
		this.setState({ lessonPopupVisible: !this.state.lessonPopupVisible })
	}

	renderLesson = () => {
		const { lesson } = this.state
		return (
			<Fragment>
				<StudentNextLessonView
					onPress={() => this.lessonPress()}
					testID="lessonRowTouchable"
					lesson={lesson}
					loading={this.state.loading}
				/>
				<LessonPopup
					visible={this.state.lessonPopupVisible}
					item={lesson}
					onPress={this.lessonPress}
					testID="lessonPopup"
					navigation={this.props.navigation}
				/>
			</Fragment>
		)
	}

	render() {
		return (
			<ScrollView>
				<View style={styles.container}>
					<View style={styles.settingsIcon}>
						<TouchableOpacity
							onPress={this._navigateToSettings.bind(this)}
						>
							<Icon name="settings" type="material" size={24} />
						</TouchableOpacity>
					</View>
					<View testID="welcomeHeader" style={styles.welcomeHeader}>
						<Image
							style={styles.profilePic}
							source={{
								uri:
									"https://images.unsplash.com/photo-1535643302794-19c3804b874b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2134&q=80"
							}}
						/>
						<Text style={styles.welcomeText}>
							{strings("teacher.home.welcome", {
								name: this.props.user["name"]
							})}
						</Text>
					</View>
					<ShadowRect style={styles.schedule}>
						<Text style={styles.rectTitle} testID="schedule">
							{strings("teacher.home.next_lesson")}
						</Text>
						{this.renderLesson()}
					</ShadowRect>

					<View style={styles.fullScheduleView}>
						<TouchableHighlight
							underlayColor="lightgray"
							onPress={() => {
								this.props.navigation.navigate("Schedule")
							}}
						>
							<Text style={styles.fullSchedule}>
								{strings("teacher.home.full_schedule")}
							</Text>
						</TouchableHighlight>
						<Icon
							size={20}
							color="rgb(12, 116, 244)"
							name="ios-arrow-dropleft-circle"
							type="ionicon"
						/>
					</View>
					<ShadowRect>
						<View style={{ flex: 1, flexDirection: "row" }}>
							<Text
								testID="monthlyAmount"
								style={styles.rectTitle}
							>
								{strings("student.home.payments")}
							</Text>
							<View
								style={{
									flex: 1,
									alignItems: "flex-end",
									marginRight: "auto"
								}}
							>
								<Icon
									name="arrow-back"
									type="material"
									size={20}
								/>
							</View>
						</View>
						<StudentPayments
							sum={this.props.user.balance}
							payments={this.state.payments.slice(0, 2)}
							loading={this.state.loading}
						/>
					</ShadowRect>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: 20,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	settingsIcon: {
		position: "absolute",
		right: MAIN_PADDING
	},
	profilePic: {
		width: 44,
		height: 44,
		borderRadius: 22,
		marginBottom: 16
	},
	welcomeHeader: {
		alignSelf: "center",
		alignItems: "center",
		marginBottom: 20
	},
	welcomeText: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	hoursStyle: { marginBottom: -8 },
	rectTitle: {
		fontSize: 16,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start"
	},
	lessonRow: {
		marginTop: 12
	},
	fullScheduleView: {
		flexDirection: "row",
		flex: 1,
		alignSelf: "center",
		alignItems: "center",
		marginTop: 4,
		marginBottom: 24
	},
	fullSchedule: {
		color: "rgb(12, 116, 244)",
		marginTop: -4,
		fontWeight: "bold",
		marginRight: 8
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Home)
