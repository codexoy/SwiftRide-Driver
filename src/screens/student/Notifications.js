import React, { Fragment } from "react"
import {
	View,
	Text,
	TouchableHighlight,
	StyleSheet,
	FlatList,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING, NAME_LENGTH } from "../../consts"
import { Dropdown } from "react-native-material-dropdown"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"
import Notification from "../../components/Notification"
import moment from "moment"
import LessonPopup from "../../components/LessonPopup"
import Hours from "../../components/Hours"
import ShowReceipt from "../../components/ShowReceipt"

export class Notifications extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		super(props)
		this.filterOptions = [
			{
				value: "lessons/",
				label: strings("notifications.scheduled_lessons")
			},
			{
				value: "lessons/payments",
				label: strings("notifications.payments")
			}
		]
		this.state = {
			items: [],
			page: 1,
			nextUrl: "",
			filter: "",
			refreshing: false,
			visible: []
		}

		this._dropdownChange = this._dropdownChange.bind(this)
	}

	_onNavigationFocus = () => {
		this.setState(
			{
				filter:
					this.props.navigation.getParam("filter") ||
					this.filterOptions[0]["value"],
				loading: true,
				page: 1,
				nextUrl: ""
			},
			() => {
				this._getItems()
			}
		)
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._onNavigationFocus()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	onRefresh = () => {
		this.setState({ page: 1, nextUrl: "", refreshing: true }, () => {
			this._getItems()
		})
	}

	_getItems = async (append = false) => {
		const resp = await this.props.fetchService.fetch(
			"/" +
				this.state.filter +
				"?limit=10&order_by=created_at desc&is_approved=true&creator_id=" +
				this.props.user.my_teacher.user.id +
				"&page=" +
				this.state.page,
			{
				method: "GET"
			}
		)
		let newValue = resp.json.data
		if (append) {
			newValue = [...this.state.items, ...newValue]
		}
		this.setState({
			items: newValue,
			nextUrl: resp.json["next_url"],
			loading: false,
			refreshing: false
		})
	}

	renderPayment = ({ item, index }) => {
		const date = moment
			.utc(item.created_at)
			.local()
			.format("DD.MM.YY, HH:mm")
		return (
			<Notification
				style={styles.notification}
				key={`payment${item.id}`}
				leftSide={<Text style={styles.amount}>{item.amount}â‚ª</Text>}
				basic={
					<Fragment>
						<Text style={styles.basic}>{date}</Text>
						<ShowReceipt
							item={item}
							dispatch={this.props.dispatch}
							user={this.props.user}
							style={{ alignSelf: "flex-start" }}
						/>
					</Fragment>
				}
			/>
		)
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderLesson = ({ item, index }) => {
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity onPress={() => this.lessonPress(item)}>
					<Notification
						style={styles.notification}
						key={`lesson${item.id}`}
						basic={
							<Text style={styles.basic}>
								{this.props.user.my_teacher.user.name.slice(
									0,
									NAME_LENGTH
								)}{" "}
								{strings("notifications.teacher_scheduled")}
							</Text>
						}
						leftSide={
							<View>
								<Text style={styles.lessonDate}>
									{moment
										.utc(item.date)
										.local()
										.format("DD.MM")}
								</Text>
								<Text style={styles.lessonHour}>
									<Hours
										duration={item.duration}
										date={item.date}
									/>
								</Text>
							</View>
						}
					/>
					<LessonPopup
						visible={visible}
						item={item}
						onPress={this.lessonPress}
						testID="lessonPopup"
						navigation={this.props.navigation}
					/>
				</TouchableOpacity>
			</Fragment>
		)
	}

	_renderItems = () => {
		if (this.state.loading) {
			return <StudentsLoader />
		}
		let render = this.renderLesson
		let type = "lesson"
		if (this.state.filter.includes("payments")) {
			render = this.renderPayment
			type = "payment"
		}
		return (
			<FlatList
				data={this.state.items}
				renderItem={render}
				onEndReached={this.endReached.bind(this)}
				keyExtractor={item => `${type}${item.id}`}
				ListEmptyComponent={this._renderEmpty}
				extraData={this.state.visible}
				onRefresh={() => this.onRefresh()}
				refreshing={this.state.refreshing}
			/>
		)
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1,
				nextUrl: ""
			},
			() => {
				this._getItems(true)
			}
		)
	}

	_dropdownChange = (value, index, data) => {
		this.setState(
			{
				page: 1,
				nextUrl: "",
				filter: value
			},
			() => {
				this._getItems()
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="notifications"
			text={strings("empty_notifications")}
			style={styles.empty}
		/>
	)

	render() {
		return (
			<View style={styles.container}>
				<View testID="NotificationsView" style={styles.notifications}>
					<PageTitle
						style={styles.title}
						title={strings("tabs.notifications_title")}
						leftSide={
							<View>
								<Dropdown
									containerStyle={styles.dropdown}
									value={this.state.filter}
									data={this.filterOptions}
									onChangeText={this._dropdownChange}
									dropdownMargins={{ min: 0, max: 20 }}
									dropdownOffset={{ top: 0, left: 0 }}
									pickerStyle={{ marginTop: 60 }}
								/>
							</View>
						}
					/>
					<View style={styles.items}>{this._renderItems()}</View>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: MAIN_PADDING,
		paddingLeft: MAIN_PADDING,
		marginTop: 20
	},
	items: {
		marginTop: 20
	},
	notification: {},
	title: { marginBottom: 0 },
	dropdown: {
		alignSelf: "flex-end",
		width: 160,
		marginTop: 4
	},
	empty: {
		marginTop: 100
	},
	amount: {
		marginTop: 12,
		fontSize: 20,
		fontWeight: "bold"
	},
	basic: {
		marginTop: 10,
		fontSize: 18
	},
	lessonDate: {
		fontSize: 22,
		fontFamily: "Assistant-Light"
	},
	lessonHour: {
		fontWeight: "bold",
		fontSize: 14,
		color: "rgb(12, 116, 244)"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Notifications)
