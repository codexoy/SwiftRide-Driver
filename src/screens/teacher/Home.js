import React, { Fragment } from "react"
import {
	ScrollView,
	View,
	Text,
	StyleSheet,
	TouchableHighlight,
	TouchableOpacity,
	FlatList
} from "react-native"
import { connect } from "react-redux"
import ShadowRect from "../../components/ShadowRect"
import UserWithPic from "../../components/UserWithPic"
import { strings } from "../../i18n"
import Row from "../../components/Row"
import Separator from "../../components/Separator"
import { Icon } from "react-native-elements"
import Hours from "../../components/Hours"
import LessonPopup from "../../components/LessonPopup"
import { MAIN_PADDING, colors, NAME_LENGTH } from "../../consts"
import { getPayments } from "../../actions/lessons"
import EmptyState from "../../components/EmptyState"
import LessonsLoader from "../../components/LessonsLoader"
import PaymentsLoader from "../../components/PaymentsLoader"
import { NavigationActions } from "react-navigation"
import { getUserImage, uploadUserImage } from "../../actions/utils"
import UploadProfileImage from "../../components/UploadProfileImage"
import moment from "moment"

export class Home extends React.Component {
	static navigationOptions = () => {
		return {
			title: "home",
			tabBarLabel: strings("tabs.home"),
			tabBarAccessibilityLabel: strings("tabs.home"),
			tabBarTestID: "HomeTab"
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			currentLesson: null,
			nextLesson: null,
			payments: [],
			sum: 0,
			visible: [],
			loading: true
		}
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._sendRequests()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	_sendRequests = async () => {
		await this._getLessons()
		await this._getPayments()
		this.setState({ loading: false })
	}

	_getLessons = async () => {
		const now = new Date().toISOString()
		const endOfDay = moment
			.utc()
			.endOf("day")
			.toISOString()
		const currentLessonResp = await this.props.fetchService.fetch(
			"/lessons/?limit=1&is_approved=true&date=ge:" +
				now +
				"&date=le:" +
				endOfDay,
			{ method: "GET" }
		)
		const nextLessonResp = await this.props.fetchService.fetch(
			"/lessons/?limit=2&is_approved=true&date=ge:" + now,
			{ method: "GET" }
		)
		let nextLesson = null,
			currentLesson = null

		if (nextLessonResp.json["data"].length > 0) {
			if (currentLessonResp.json["data"].length == 0) {
				// no current lesson (no lessons today)
				nextLesson = nextLessonResp.json["data"][0]
			} else {
				currentLesson = currentLessonResp.json["data"][0]
				nextLesson = nextLessonResp.json["data"][1]
			}
		}

		this.setState({
			currentLesson,
			nextLesson
		})
	}

	_getPayments = async () => {
		const payments = await getPayments(this.props.fetchService)
		this.setState({
			payments: payments.payments,
			sum: payments.sum
		})
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderLesson = item => {
		if (!item) {
			return (
				<Text
					style={{
						...styles.lessonRow,
						...styles.noLesson
					}}
				>
					{strings("teacher.home.no_lesson")}
				</Text>
			)
		}
		let student = strings("teacher.no_student_applied")
		let user = null
		if (item.student) {
			student = item.student.user.name
			user = item.student.user
		}
		const date = item.date
		let meetup = strings("not_set")
		if (item.meetup_place) meetup = item.meetup_place.name
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity
					onPress={() => this.lessonPress(item)}
					testID="lessonRowTouchable"
				>
					<Row
						key={`item${item.id}`}
						style={styles.lessonRow}
						leftSide={
							<Hours
								style={styles.hours}
								duration={item.duration}
								date={date}
							/>
						}
					>
						<UserWithPic
							name={student}
							user={user}
							extra={
								<View style={{ alignItems: "flex-start" }}>
									<Text style={styles.places}>
										{strings("teacher.new_lesson.meetup")}:{" "}
										{meetup}
									</Text>
								</View>
							}
							nameStyle={styles.nameStyle}
							imageContainerStyle={styles.placesImage}
						/>
					</Row>
				</TouchableOpacity>
				<LessonPopup
					visible={visible}
					item={item}
					onPress={this.lessonPress}
					testID="lessonPopup"
					navigation={this.props.navigation}
				/>
			</Fragment>
		)
	}

	renderPaymentItem = ({ item, index }) => {
		let firstItemStyles = {}
		if (index == 0) {
			firstItemStyles = { marginTop: 0 }
		}
		return (
			<Row
				style={{ ...styles.paymentRow, ...firstItemStyles }}
				leftSide={
					<Text style={styles.amountOfStudent}>{item.amount}₪</Text>
				}
				key={`payment${item.id}`}
			>
				<UserWithPic
					user={item.student.user}
					nameStyle={styles.nameStyle}
				/>
			</Row>
		)
	}

	_renderEmpty = type => (
		<EmptyState
			image={type}
			text={strings(`empty_${type}`)}
			imageSize="small"
		/>
	)

	_navigateToSettings = () => {
		this.props.navigation.navigate("Settings")
	}

	_renderLessons = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<LessonsLoader />
				</View>
			)
		}

		return (
			<Fragment>
				<Text style={styles.rectTitle} testID="schedule">
					{strings("teacher.home.current_lesson")}
				</Text>
				{this.renderLesson(this.state.currentLesson)}
				<Separator />
				<Text style={styles.rectTitle}>
					{strings("teacher.home.next_lesson")}
				</Text>
				{this.renderLesson(this.state.nextLesson)}
			</Fragment>
		)
	}

	_renderPayments = () => {
		if (this.state.loading) {
			return (
				<View style={styles.listLoader}>
					<PaymentsLoader />
				</View>
			)
		}
		let sumColor = colors.green
		if (this.state.sum < 0) sumColor = "red"
		let list
		if (this.state.payments.length == 0) {
			list = this._renderEmpty("payments")
		} else {
			list = (
				<FlatList
					data={this.state.payments.slice(0, 2)}
					renderItem={this.renderPaymentItem}
					keyExtractor={item => `payment${item.id}`}
				/>
			)
		}
		return (
			<Fragment>
				<View style={styles.amountView}>
					<Text style={{ ...styles.amount, color: sumColor }}>
						{this.state.sum}₪
					</Text>
					<TouchableOpacity
						onPress={() =>
							this.props.navigation.navigate("AddPayment")
						}
					>
						<Text style={styles.addPayment}>
							{strings("teacher.home.add_payment")}
						</Text>
					</TouchableOpacity>
				</View>
				<Separator />
				{list}
			</Fragment>
		)
	}

	render() {
		return (
			<ScrollView>
				<View style={styles.container}>
					<View style={styles.settingsIcon}>
						<TouchableOpacity
							onPress={this._navigateToSettings.bind(this)}
						>
							<Icon name="settings" type="material" size={24} />
						</TouchableOpacity>
					</View>
					<View testID="welcomeHeader" style={styles.welcomeHeader}>
						<UploadProfileImage
							style={styles.profilePic}
							image={getUserImage(this.props.user)}
							upload={async source => {
								await this.props.dispatch(
									uploadUserImage(source)
								)
							}}
						/>
						<Text style={styles.welcomeText}>
							{strings("teacher.home.welcome", {
								name: this.props.user.name.slice(0, NAME_LENGTH)
							})}
						</Text>
					</View>
					<ShadowRect style={styles.schedule}>
						{this._renderLessons()}
					</ShadowRect>

					<View style={styles.fullScheduleView}>
						<TouchableHighlight
							underlayColor="lightgray"
							onPress={() => {
								this.props.navigation.navigate("Schedule")
							}}
						>
							<Text style={styles.fullSchedule}>
								{strings("teacher.home.full_schedule")}
							</Text>
						</TouchableHighlight>
						<Icon
							size={20}
							color="rgb(12, 116, 244)"
							name="ios-arrow-dropleft-circle"
							type="ionicon"
						/>
					</View>
					<ShadowRect>
						<View style={{ flex: 1, flexDirection: "row" }}>
							<Text
								testID="monthlyAmount"
								style={styles.rectTitle}
							>
								{strings("teacher.home.monthly_amount")}
							</Text>
							<TouchableOpacity
								style={{
									flex: 1,
									alignItems: "flex-end",
									marginRight: "auto"
								}}
								onPress={() =>
									this.props.navigation.navigate({
										routeName: "Notifications",
										params: {},
										action: NavigationActions.navigate({
											routeName: "Main",
											params: {
												filter: "lessons/payments"
											}
										})
									})
								}
							>
								<Icon
									name="arrow-back"
									type="material"
									size={20}
								/>
							</TouchableOpacity>
						</View>
						{this._renderPayments()}
					</ShadowRect>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: 20,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	schedule: { minHeight: 220 },
	welcomeHeader: {
		alignSelf: "center",
		alignItems: "center",
		marginBottom: 20
	},
	settingsIcon: {
		position: "absolute",
		right: MAIN_PADDING
	},
	welcomeText: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	profilePic: {
		width: 44,
		height: 44,
		borderRadius: 22,
		marginBottom: 16
	},
	rectTitle: {
		fontSize: 16,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start"
	},
	lessonRow: {
		marginTop: 12
	},
	hours: { marginTop: 8 },
	fullScheduleView: {
		flexDirection: "row",
		flex: 1,
		alignSelf: "center",
		alignItems: "center",
		marginTop: 4,
		marginBottom: 24
	},
	fullSchedule: {
		color: "rgb(12, 116, 244)",
		marginTop: -4,
		fontWeight: "bold",
		marginRight: 8
	},
	places: {
		fontSize: 14,
		color: "gray"
	},
	placesImage: { marginTop: 8 },
	amountView: {
		marginTop: 16,
		alignSelf: "center"
	},
	amount: {
		fontFamily: "Assistant-Light",
		fontSize: 44
	},
	addPayment: {
		color: "rgb(12, 116, 244)",
		fontWeight: "bold",
		marginTop: 16,
		alignSelf: "center"
	},
	amountOfStudent: {
		marginTop: 4
	},
	nameStyle: {
		marginTop: 4,
		marginLeft: -2
	},
	paymentRow: {
		marginTop: 12
	},
	listLoader: { marginTop: 20, alignSelf: "center" },
	noLesson: {
		fontSize: 20,
		alignSelf: "center"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Home)
