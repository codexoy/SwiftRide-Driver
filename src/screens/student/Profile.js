import React from "react"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import StudentProfile from "../../components/StudentProfile"
import EditStudent from "../../components/EditStudent"
import Notifications from "../teacher/Notifications"
import { createStackNavigator } from "react-navigation"
import Topics from "../Topics"

export class Profile extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		// only here for the test suite to work
		super(props)
	}

	render() {
		return <StudentProfile {...this.props} />
	}
}

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService,
		error: state.error
	}
}
const profileScreen = connect(mapStateToProps)(Profile)
const editScreen = connect(mapStateToProps)(EditStudent)
const profile = createStackNavigator(
	{
		Main: profileScreen,
		Topics,
		EditStudent: editScreen,
		Payments: Notifications
	},
	{
		initialRouteName: "Main",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
export default profile
