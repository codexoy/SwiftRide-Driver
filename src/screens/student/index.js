import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import { tabBarOptions } from "../consts"
import Home from "./Home"
import Notifications from "./Notifications"
import Lesson from "./Lesson"
import Schedule from "./Schedule"
import Profile from "./Profile"
import Settings from "../Settings"
import StudentProfile from "../student/Profile"

export default createBottomTabNavigator(
	{
		Home: createStackNavigator(
			{
				Main: createStackNavigator(
					{ Home, Lesson },
					{
						initialRouteKey: "Home",
						headerMode: "none"
					}
				),
				Settings
			},
			{
				initialRouteName: "Main",
				mode: "modal",
				headerMode: "none",
				navigationOptions: {
					headerVisible: false
				}
			}
		),
		Notifications: createStackNavigator(
			{ Main: Notifications, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		AddLesson: Lesson,
		Schedule: createStackNavigator(
			{ Main: Schedule, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		StudentProfile: Profile
	},
	tabBarOptions
)
