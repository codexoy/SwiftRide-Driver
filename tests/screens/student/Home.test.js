import "react-native"
import React from "react"
import FetchService from "../../../src/services/Fetch"
import renderer from "react-test-renderer"

import { Home } from "../../../src/screens/student/Home"

const fetchService = new FetchService()
jest.useFakeTimers()
describe("Home", () => {
	test("view renders correctly", () => {
		const user = { name: "test" }
		const student = { user }
		fetch.mockResponseSuccess(
			JSON.stringify({
				data: [
					{
						student,
						dropoff_place: { name: "t" },
						meetup_place: { name: "a" }
					}
				]
			})
		)
		fetch.mockResponseSuccess(
			JSON.stringify({ data: [{ amount: 100, student }] })
		)
		const tree = renderer
			.create(<Home user={user} fetchService={fetchService} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
