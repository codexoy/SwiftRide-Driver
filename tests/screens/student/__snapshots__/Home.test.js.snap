// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view renders correctly 1`] = `
<ScrollViewMock>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
        "paddingLeft": 26,
        "paddingRight": 26,
      }
    }
  >
    <View
      style={
        Object {
          "position": "absolute",
          "right": 26,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Themed.Icon
          name="settings"
          size={24}
          type="material"
        />
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "marginBottom": 20,
        }
      }
      testID="welcomeHeader"
    >
      <UploadProfileImage
        image="https://res.cloudinary.com/hsmkon3br/image/upload/c_thumb,g_face,h_80,w_80/v1554188012/678099-profile-filled-256.png"
        style={
          Object {
            "borderRadius": 22,
            "height": 44,
            "marginBottom": 16,
            "width": 44,
          }
        }
        upload={[Function]}
      />
      <Text
        style={
          Object {
            "fontFamily": "Assistant-Light",
            "fontSize": 24,
          }
        }
      >
        צהריים טובים, student
      </Text>
    </View>
    <ShadowRect>
      <Text
        style={
          Object {
            "alignSelf": "flex-start",
            "color": "rgb(121, 121, 121)",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="schedule"
      >
        שיעור הבא
      </Text>
      <StudentNextLessonView
        lesson={
          Object {
            "dropoff_place": "t",
            "meetup_place": "a",
            "student": Object {
              "name": "test",
            },
          }
        }
        lessonPress={[Function]}
        loading={false}
        testID="lessonRowTouchable"
      />
      <LessonPopup
        item={
          Object {
            "dropoff_place": "t",
            "meetup_place": "a",
            "student": Object {
              "name": "test",
            },
          }
        }
        navigation={
          Object {
            "addListener": [MockFunction] {
              "calls": Array [
                Array [
                  "willFocus",
                  [Function],
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
          }
        }
        onPress={[Function]}
        testID="lessonPopup"
        visible={false}
      />
    </ShadowRect>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(12, 116, 244)",
            "fontWeight": "bold",
            "marginRight": 8,
            "marginTop": -4,
          }
        }
      >
        לו״ז מלא
      </Text>
      <Themed.Icon
        color="rgb(12, 116, 244)"
        name="ios-arrow-dropleft-circle"
        size={20}
        type="ionicon"
      />
    </TouchableOpacity>
    <ShadowRect>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
              "color": "rgb(121, 121, 121)",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
          testID="monthlyAmount"
        >
          תשלומים
        </Text>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "marginRight": "auto",
            }
          }
        >
          <Themed.Icon
            name="arrow-back"
            size={20}
            type="material"
          />
        </View>
      </TouchableOpacity>
      <StudentPayment
        dispatch={[MockFunction]}
        loading={false}
        payments={
          Array [
            Object {
              "amount": 100,
              "student": Object {
                "name": "test",
              },
            },
          ]
        }
        user={
          Object {
            "name": "student",
          }
        }
      />
    </ShadowRect>
  </View>
</ScrollViewMock>
`;
